src/macaron/slsa_analyzer/git_service/github.py

```diff
@@ -56,3 +58,34 @@ def clone_repo(self, clone_dir: str, url: str) -> None:
             If there is an error cloning the repo.
         """
         git_url.clone_remote_repo(clone_dir, url)
+
+    def check_out_repo(self, git_obj: Git, branch: str, digest: str, offline_mode: bool) -> Git:
+        """Checkout the branch and commit specified by the user of a repository.
+
+        Parameters
+        ----------
+        git_obj : Git
+            The Git object for the repository to check out.
+        branch : str
+            The branch to check out.
+        digest : str
+            The sha of the commit to check out.
+        offline_mode: bool
+            If true, no fetching is performed.
+
+        Returns
+        -------
+        Git
+            The same Git object from the input.
+
+        Raises
+        ------
+        RepoError
+            If there is error while checkout the specific branch and digest.
+        """
+        if not git_url.check_out_repo_target(git_obj, branch, digest, offline_mode):
+            raise RepoError(
```

Comment:
I'm not sure why we need to raise an exception here. The issue seems to be isolated enough to properly log the error and return `None` if check out fails.

Reply:
Even though it's isolated, I want to keep this consistent with what we have in `gitlab.py` where the exception is the only way to communicate that the checkout failed.